lib/modules/Server.js

```diff
@@ -13,6 +13,31 @@ import actions from './actions';
 import { createQuery } from './query';
 import { METHODS } from './router';
 
+export const dispatchInitialShell = async (ctx, dispatch) => {
```

Comment:
this seems like a special function, one that plays well with `dispatchBeforeNavigation`. it would be nice to through this into a separate module altogether, like `plugins`. we could then expand on plugins and do more stuff with it in the future.